File: is52038b-labs/16/cpp/LCG2935256Test.cpp
#include <cppunit/extensions/HelperMacros.h>
#include <cppunit/CompilerOutputter.h>
#include <cppunit/extensions/TestFactoryRegistry.h>
#include <cppunit/ui/text/TestRunner.h>

#include "LCG.hpp"

class LCG2935256Test : public CppUnit::TestFixture {
  CPPUNIT_TEST_SUITE(LCG2935256Test);
  CPPUNIT_TEST(testSequence);
  CPPUNIT_TEST(testNext);
  CPPUNIT_TEST(testSeed);
  CPPUNIT_TEST_SUITE_END();
public:
  void setUp() { };
  void tearDown() { };

  void testSequence();
  void testNext();
  void testSeed();
};

void LCG2935256Test::testSequence() {
  LCG g = LCG(29, 35, 256, 1);
  {
    uint32_t z[10] = {64, 99, 90, 85, 196, 87, 254, 233, 136, 139};
    for (int i = 0; i < 10; i++) {
      CPPUNIT_ASSERT_EQUAL_MESSAGE("unexpected next", z[i], g.next());
    }
  }
  g = LCG(29, 35, 256, 2);
  {
    uint32_t z[10] = {93, 172, 159, 38, 113, 240, 83, 138, 197, 116};
    for (int i = 0; i < 10; i++) {
      CPPUNIT_ASSERT_EQUAL_MESSAGE("unexpected next", z[i], g.next());
    }
  }
  g = LCG(29, 35, 256, 3);
  {
    uint32_t z[10] = {122, 245, 228, 247, 30, 137, 168, 43, 2, 93};
    for (int i = 0; i < 10; i++) {
      CPPUNIT_ASSERT_EQUAL_MESSAGE("unexpected next", z[i], g.next());
    }
  }
}

void LCG2935256Test::testNext() {
  LCG g = LCG(29, 35, 256, 0);
  uint32_t z = 0;
  for (int i = 0; i < 256; i++) {
    z = (29*z + 35) % 256;
    uint32_t x = g.next();
    CPPUNIT_ASSERT_EQUAL_MESSAGE("unexpected next", z, x);
  }
}

void LCG2935256Test::testSeed() {
  LCG g = LCG(29, 35, 256, 11);
  g.seed(1);
  CPPUNIT_ASSERT_EQUAL_MESSAGE("seed appears not to have seeded", 64u, g.next());
}

CPPUNIT_TEST_SUITE_NAMED_REGISTRATION(LCG2935256Test, "2935256");
CPPUNIT_REGISTRY_ADD_TO_DEFAULT("2935256");

File: is52038b-labs/16/java/LCG2935256Test.java
import static java.util.Arrays.asList;
import java.util.List;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;
import java.util.Random;
import org.junit.Test;
import org.junit.runner.JUnitCore;
import org.junit.runner.Result;

public class LCG2935256Test {
    @Test
    public void testSequence() {
        LCG g = new LCG(29, 35, 256, 1);
        {
            long z[] = {64, 99, 90, 85, 196, 87, 254, 233, 136, 139};
            for (int i = 0; i < 10; i++) {
                assertEquals("unexpected next", z[i], g.next());
            }
        }
        g = new LCG(29, 35, 256, 2);
        {
            long z[] = {93, 172, 159, 38, 113, 240, 83, 138, 197, 116};
            for (int i = 0; i < 10; i++) {
                assertEquals("unexpected next", z[i], g.next());
            }
        }
        g = new LCG(29, 35, 256, 3);
        {
            long z[] = {122, 245, 228, 247, 30, 137, 168, 43, 2, 93};
            for (int i = 0; i < 10; i++) {
                assertEquals("unexpected next", z[i], g.next());
            }
        }
    }
    @Test
    public void testNext() {
        LCG g = new LCG(29, 35, 256, 0);
        long z = 0;
        for (int i = 0; i < 256; i++) {
            z = (29*z + 35) % 256;
            long x = g.next();
            assertEquals("unexpected next", z, x);
        }
    };
    @Test
    public void testSeed() {
        LCG g = new LCG(29, 35, 256, 11);
        g.seed(1);
        assertEquals("seed appears not to have seeded", 64, g.next());
    };
}
